### PR TITLE
Code reviews

### PR DESCRIPTION
**Pull request by pepibumur**
### Short description 📝
This PR documents what we expect from reviewers when they review PRs in any project of the Tuist organization.

### CODE REVIEW

**Comment by pepibumur** [comment]
@tuist/core would you add any other item?

File: docs/contribution/code-reviews.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+Inconsistencies complicate maintenance, and therefore we should avoid them.
+If there's an approach to output messages to the user,
+or report errors,
+we should stick to that.
+If the author disagrees with the project's standards,
+suggest them to open an issue where we can discuss them further.
+
+**✅ Tests:** Tests allow changing code with confidence.
+The code on pull requests should be tested,
+and tests should be good.
```
Comment (ollieatkinson) [inline docs/contribution/code-reviews.md]:
```suggestion
and all tests should pass.
```

File: docs/contribution/code-reviews.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+A good test is a test that consistently produces the same result and that it's easy to understand and maintain.
+Reviewers spend most of the review time in the implementation code,
+but tests are equally important because they are code too.
+
+**⚠️ Breaking changes:** Breaking changes are a bad user experience from Tuist.
```
Comment (ollieatkinson) [inline docs/contribution/code-reviews.md]:
```suggestion
**⚠️ Breaking changes:** Breaking changes are a bad user experience for users of Tuist.
```

File: docs/contribution/code-reviews.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+Contributions should avoid introducing breaking changes unless it's strictly necessary.
+There are many language feature that we can leverage to evolve the interface of Tuist without resorting to them.
```
Comment (ollieatkinson) [inline docs/contribution/code-reviews.md]:
```suggestion
There are many language features that we can leverage to evolve the interface of Tuist without resorting to a breaking change.
```

File: docs/contribution/code-reviews.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+Whether a change is breaking or not might not be obvious.
```
Comment (ollieatkinson) [inline docs/contribution/code-reviews.md]: We could also suggest to check changes against the fixtures to ensure old manifests are still compatible.

File: docs/contribution/code-reviews.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+It requires putting ourselves in the user's shoes and imagine how the changes would impact them.
+
+**📝 Documentation:** As the project grows and we continue to add more features,
+keeping the documentation up to date is crucial for developers to adopt them.
+Moreover,
+it's a very valuable asset for new adopters that are giving Tuist a try.
+Pull request that change the user interface of Tuist,
```
Comment (ollieatkinson) [inline docs/contribution/code-reviews.md]:
```suggestion
Pull requests that change the user interface of Tuist,
```

File: docs/contribution/code-reviews.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+for example adding support for a new argument,
+must include documentation.
+
+**📝 Changelog:** The `CHANGELOG.md` file to put together a list of changes that are released with new versions.
```
Comment (ollieatkinson) [inline docs/contribution/code-reviews.md]:
```suggestion
**📝 Changelog:** The `CHANGELOG.md` file contains a list of changes that are released with new versions.
```

**Comment by pepibumur** [comment]
Thanks @ollieatkinson for the review, very appreciated 😄

**Pull request merged**